File: bindings/nodejs/examples/message_examples/1-create-account.js
/**
 * This example creates a new database and account
 */

require('dotenv').config();

async function run() {
    const { AccountManagerForMessages } = require('@iota/wallet');

    const manager = new AccountManagerForMessages({
        storagePath: './alice-database',
    });

    try {
        await manager.setStrongholdPassword(process.env.SH_PASSWORD);
        await manager.storeMnemonic();

        const account = await manager.createAccount({
            // todo replace with https://api.lb-0.h.chrysalis-devnet.iota.cafe when the new faucet is working
            clientOptions: {
                node: { url: 'https://api.lb-0.h.chrysalis-devnet.iota.cafe' },
                localPow: true,
            },
            alias: 'Alice',
        });
        console.log('Account created:', account);
    } catch (error) {
        console.log('Error: ' + error);
    }
}

run();

File: bindings/python/native/example/1-create-account.py
from iota_wallet import IotaWallet

# This example creates a new database and account

client_options = {
    'primaryNode': {'url': 'https://api.lb-0.h.chrysalis-devnet.iota.cafe'},
    'localPow': True,
}
secret_manager = ("flame fever pig forward exact dash body idea link scrub tennis minute " +
          "surge unaware prosper over waste kitten ceiling human knife arch situate civil")

wallet = IotaWallet('./alice-database', client_options, secret_manager)
account = wallet.create_account('Alice')
print(account)
